pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTLiteral.java

```diff
@@ -252,6 +258,100 @@ public boolean isSingleCharacterStringLiteral() {
+     * removing incidental white space surrounding the text block and interpreting
+     * escape sequences.
+     */
+    @Experimental
+    public String getTextBlockContent() {
+        if (!isTextBlock()) {
+            throw new IllegalArgumentException("This is not a text block");
+        }
+
+        int start = determineContentStart(getImage());
+        String content = getImage().substring(start, getImage().length() - TEXTBLOCK_DELIMITER.length());
+        // normalize line endings to LF
+        content = content.replaceAll("\r\n|\r", "\n");
+
+        int prefixLength = Integer.MAX_VALUE;
+        List<String> lines = Arrays.asList(content.split("\\n"));
+        for (int i = 0; i < lines.size(); i++) {
+            String line = lines.get(i);
+            // compute common prefix
+            if (!StringUtils.isAllBlank(line) || i == lines.size() - 1) {
+                prefixLength = Math.min(prefixLength, countLeadingWhitespace(line));
+            }
+        }
+        if (prefixLength == Integer.MAX_VALUE) {
+            // common prefix not found
+            prefixLength = 0;
+        }
+        StringBuilder sb = new StringBuilder(content.length());
+        for (int i = 0; i < lines.size(); i++) {
+            String line = lines.get(i);
+            // remove common whitespace prefix
+            if (!StringUtils.isAllBlank(line) && line.length() >= prefixLength) {
+                line = line.substring(prefixLength);
+            }
+            line = removeTrailingWhitespace(line);
+            sb.append(line);
+
+            boolean isLastLine = i == lines.size() - 1;
+            boolean isFirstLine = i == 0;
+            if (!isLastLine || !isFirstLine && !StringUtils.isAllBlank(line)) {
+                sb.append('\n');
+            }
+        }
+        String result = sb.toString();
+
+        // interpret escape sequences "\NL" "n","t","b","r","f", "s", "\"", "\'"
+        result = result
+                    .replaceAll("\\\\\n", "")
+                    .replaceAll("\\\\n", "\n")
+                    .replaceAll("\\\\t", "\t")
+                    .replaceAll("\\\\b", "\b")
+                    .replaceAll("\\\\r", "\r")
+                    .replaceAll("\\\\f", "\f")
+                    .replaceAll("\\\\s", " ")
+                    .replaceAll("\\\\\"", "\"")
+                    .replaceAll("\\\\'", "'");
+        return result;
+    }
+
+    private static int determineContentStart(String s) {
+        int start = TEXTBLOCK_DELIMITER.length(); // this is the opening delimiter
+        boolean lineTerminator = false;
+        // the content begins after at the first character after the line terminator
+        // of the opening delimiter
+        while (start < s.length() && Character.isWhitespace(s.charAt(start))) {
+            if (s.charAt(start) == '\r' || s.charAt(start) == '\n') {
+                lineTerminator = true;
```

Comment:
Does this work with CRLF delimited strings? It looks like it's not going past the LF. Maybe you just need to call this method on the normalized string, and stop caring about CR.

Also maybe it would be clearer to just `return start + 1;` here instead of breaking on the next iteration.

Reply:
There was a problem with a corner case: If the content starts with LF, so that the literal looks like this: `"""<LF><LF>foo<LF>"""`, then we would have read passed the 2nd LF and missed that... I've fixed it.